### PR TITLE
feat: add optional httpsOptions parameter to NodeHttpTransport

### PR DESCRIPTION
**Pull request by notmedia**
Hi! I'm working on [desktop gRPC client](https://github.com/getezy/ezy) using this library where I need the ability to support Self-Signed certificates for https requests. 

## Changes

Added optional httpsOptions parameter to NodeHttpTransport.

Usage example:
```ts
grpc.invoke(methodDefinition, {
      ....
      transport: NodeHttpTransport({
        ca: fs.readFileSync('path/to/self-signed-cert'),
        // or
        rejectUnauthorized: false
      }),
    });
```

## Verification

Tested on macOS, node 18

### CODE REVIEW

**Review by johanbrandhorst — COMMENTED** [review]
This looks good to me, but I'm not enough of a node expert to say whether this is the right way to add an option like this. I understand this option only applies when using HTTPS, is it still appropriate to make it a top level parameter to the constructor? CC @MarcusLongmuir.

**Comment by notmedia** [comment]
Hi @johanbrandhorst!

At line 6 we have factory which creates another factory for instantiating Node transport inside core of this lib.

Because all transports controlled by core and the core invokes start method there is no another way to pass httpsOptions to this method. All what we can is instantiate transport with custom httpsOptions.

Also I made this option as readonly and optional.

**Comment by johanbrandhorst** [comment]
Hi, CI is failing for unrelated reasons, can you please rebase on https://github.com/improbable-eng/grpc-web/pull/1123?

**Comment by notmedia** [comment]
Hey, I merged your branch here, but some test pipelines failing. I’ve looked to errors and not sure that my code could broke this.

I think that first we need to fix all pipelines, merge fixes to master and after that merge features.

Also I see @MarcusLongmuir does not have any activity for a long time. I'm curious is he working on this library?

**Comment by johanbrandhorst** [comment]
The test failures are just flaky tests, I have rerun them. I don't know what Marcus' level of involvement is with the project anymore, I'm afraid. I can still merge contributions on my own though :).

**Review by johanbrandhorst — APPROVED** [review]
Lets go for it

**Comment by notmedia** [comment]
Thank you!

**Comment by notmedia** [comment]
@johanbrandhorst Hey! Could you publish new version to npm?

**Comment by johanbrandhorst** [comment]
https://github.com/improbable-eng/grpc-web/pull/1112#issuecomment-1213638995 😬. I can't make a release on my own, unfortunately.

**Comment by SoleeChoi** [comment]
Would you please publish new version to npm? If it is too much to you, how about update [grpc-web-node-http-transport](https://github.com/improbable-eng/grpc-web/tree/master/client/grpc-web-node-http-transport) repository in advance?

**Comment by KodamaHQ** [comment]
why is this not published to npm yet?

**Comment by notmedia** [comment]
@SoleeChoi @ranlavanet Hey, as was mentioned above this project was abandoned from the main maintainer who has access to npm registry. So no one can publish it.

You can copy this transport implementation to your project and connect it as custom.

Or you can use this library where I had already did it - https://github.com/getezy/grpc-client

**Comment by KodamaHQ** [comment]
@notmedia Thanks for the quick response! I have done it and it works magically thank you so much for adding this feature!

**Comment by notmedia** [comment]
> @notmedia Thanks for the quick response! I have done it and it works magically thank you so much for adding this feature!

No problem, you are welcome :)

**Pull request merged**
